ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogContainer.tsx

```diff
@@ -73,7 +73,7 @@ export function BrowseFilesDialogContainer(props: BrowseFilesDialogContainerProp
 
   const fetchItems = useCallback(
     () =>
-      search(site, { ...searchParameters, path: `${currentPath}/[^/]+` }).subscribe((response) => {
+      search(site, { ...searchParameters, path: `${currentPath}/([^/]+|[^/]+/index.xml)` }).subscribe((response) => {
```

Comment:
Regexp looks strange. Not too sure how this regex works/what it tells the service to do; what exactly is the `path` argument to the search service? The `index.xml` part seems almost redundant.
```suggestion
      search(site, { ...searchParameters, path: `${currentPath}/[^/]+(index\.xml)?` }).subscribe((response) => {
```

Reply:
Updated the regex. The idea is to retrieve everything under `currentPath` with no `/` (to avoid getting files from subfolders), **but** consider pages, so files with name `index.xml` under a folder should match too. Used your suggestion and only added an `/` before `index.xml` (IDE suggested to remove escaping chars like the one before the dot in index.xml and the regex worked)

Reply:
`.` means anything so not sure if it's working because of that or because escaping is actually not needed.

Reply:
Yeah it was actually considering index{anyChar}xml, updated for it to be `${currentPath}/[^/]+(/index\\.xml)?`. That way it does return the results the way be want (first level children and `{folder}/index.xml`)